pkg/monitoring/endpoint.go

```diff
@@ -0,0 +1,307 @@
+			http.Error(w, fmt.Sprintf("protocol '%s' is not supported", protocol), http.StatusBadRequest)
+			return
+		}
+
+		command := []string{
+			"show",
+			"bgp",
+			"vrf",
+			vrf,
+			protocol,
+			"unicast",
+		}
+
+		if err := setInput(r, &command); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if err := setLongerPrefixes(r, &command); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		data = e.cli.ExecuteWithJSON(command)
+	default:
+		http.Error(w, fmt.Sprintf("request of type '%s' is not supported", requestType), http.StatusBadRequest)
+		return
+	}
+
+	result := addNodename(&data)
+	writeResponse(result, w)
+}
+
+// ShowEVPN returns result of show evpn command.
+// show evpn vni json.
+// show evpn rmac vni <all|vrf>.
+// show evpn mac vni <all|vrf>.
+// show evpn next-hops vni <all|vrf> json.
+func (e *Endpoint) ShowEVPN(w http.ResponseWriter, r *http.Request) {
+	var data []byte
+	requestType := r.URL.Query().Get("type")
+	switch requestType {
+	case "":
+		data = e.cli.ExecuteWithJSON([]string{
+			"show",
+			"evpn",
+			"vni",
+		})
+	case "rmac", "mac", "next-hops":
+		vni := r.URL.Query().Get("vni")
+		if vni == "" {
+			vni = all
+		}
+
+		data = e.cli.ExecuteWithJSON([]string{
+			"show",
+			"evpn",
+			requestType,
+			"vni",
+			vni,
+		})
+	default:
+		http.Error(w, fmt.Sprintf("request of type '%s' is not supported", requestType), http.StatusBadRequest)
+		return
+	}
+
+	result := addNodename(&data)
+	writeResponse(result, w)
+}
+
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=list
+
+// PassRequest - when called, will pass the request to all nodes and return their responses.
+func (e *Endpoint) PassRequest(w http.ResponseWriter, r *http.Request) {
+	pods := &corev1.PodList{}
+	matchLabels := &client.MatchingLabels{
+		"app.kubernetes.io/name":      "network-operator",
+		"app.kubernetes.io/component": "worker",
+	}
+
+	err := e.c.List(r.Context(), pods, matchLabels)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error listing pods: %s", err.Error()), http.StatusInternalServerError)
+		return
+	}
+
+	if len(pods.Items) == 0 {
+		http.Error(w, "error listing pods: no pods found", http.StatusInternalServerError)
+		return
+	}
+
+	query := strings.ReplaceAll(r.URL.String(), "all/", "")
+
+	s := strings.Split(r.Host, ":")
+	port := ""
+	if len(s) > 1 {
+		port = s[1]
+	}
+
+	buffer := []byte{}
+
+	var wg sync.WaitGroup
+	results := make(chan []byte, len(pods.Items))
+	errors := make(chan error, len(pods.Items))
+
+	for i := range pods.Items {
+		wg.Add(1)
+		go passToPod(&pods.Items[i], port, query, results, errors, &wg)
+	}
+
+	wg.Wait()
+	close(results)
+	close(errors)
+
+	for err := range errors {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	for result := range results {
+		buffer = append(buffer, result...)
+		buffer = append(buffer, '\n')
+	}
```

Comment:
All the results should be in JSON format, you can probably parse them into `map[string]interface{}` individually. Afterwards you can use a map of that again (hostname to answer parsed into map[string]interface{}).

This way you get meaningful JSON from the aggregate endpoint without relying on string concatenation.

Reply:
You're right. Changed this to use `map[string]json.RawMessage{}` for proper marshaling.